packages/core/src/external/carequality/ihe-gateway-v2/outbound/xca/process/error.ts

```diff
@@ -190,7 +312,69 @@ export function handleSchemaErrorResponse({
     gateway,
     operationOutcome,
     iheGatewayV2: true,
+    externalGatewayPatient: outboundRequest.externalGatewayPatient,
+    serviceDate: outboundRequest.serviceDate,
+  };
+}
+
+export function handleSchemaErrorResponseDr({
+  outboundRequest,
+  gateway,
+  text = "Schema Error",
+}: {
+  outboundRequest: OutboundDocumentRetrievalReq;
+  gateway: XCAGateway;
+  text?: string;
+}): OutboundDocumentRetrievalResp {
+  const operationOutcome: OperationOutcome = {
+    resourceType: "OperationOutcome",
+    id: outboundRequest.id,
+    issue: [
+      ...outboundRequest.documentReference.map(doc => ({
+        id: doc.metriportId,
+        severity: "error",
+        code: schemaErrorCode,
+        details: {
+          id: doc.docUniqueId,
+          text,
+        },
+      })),
+    ],
+  };
+  return {
+    id: outboundRequest.id,
+    requestChunkId: outboundRequest.requestChunkId,
+    patientId: outboundRequest.patientId,
+    timestamp: outboundRequest.timestamp,
+    requestTimestamp: outboundRequest.timestamp,
+    responseTimestamp: dayjs().toISOString(),
+    gateway,
+    operationOutcome,
+    iheGatewayV2: true,
+  };
+}
+
+export function generateOperationOutcomesForMissingDocuments(
+  idMapping: Record<string, string>,
+  processedIds: Set<string>,
+  requestId: string
+): OperationOutcome {
+  const operationOutcome: OperationOutcome = {
+    resourceType: "OperationOutcome",
+    id: requestId,
+    issue: Object.entries(idMapping)
+      .filter(([, metriportId]) => !processedIds.has(metriportId))
+      .map(([docUniqueId, metriportId]) => ({
+        id: metriportId,
+        severity: "information",
+        code: "document-not-found",
+        details: {
+          id: docUniqueId,
+          text: `Document with metriportId ${metriportId} and id ${docUniqueId} not found`,
```

Comment:
this msg is very very similar to the one on 264, but they account for slightly different cases. Would be good to reflect that in the text here